src/qibo/tomography/gate_set_tomography.py

```diff
@@ -0,0 +1,371 @@
+            raise_error(
+                ValueError,
+                f"Mismatched inputs: nqubits given as {nqubits}. {gate} is a {len(gate.qubits)}-qubit gate.",
+            )
+        gate = gate.__class__(*gate.qubits, **gate.init_kwargs)
+
+    # GST for empty circuit or with gates
+    matrix_jk = np.zeros((4**nqubits, 4**nqubits))
+    for k in range(4**nqubits):
+        circ = Circuit(nqubits, density_matrix=True)
+        circ.add(_prepare_state(k, nqubits))
+        if invert_register is not None:
+            inverted_circuit = reset_register(circ, invert_register)
+            circ.add(inverted_circuit.on_qubits(*invert_register))
+
+        if gate is not None:
+            circ.add(gate)
+
+        for j in range(4**nqubits):
+            new_circ = circ.copy()
+            measurements = _measurement_basis(j, nqubits)
+            new_circ.add(measurements)
+            if noise_model is not None and backend.name != "qibolab":
+                new_circ = noise_model.apply(new_circ)
+            expectation_val = _expectation_value(new_circ, j, nshots, backend=backend)
+            matrix_jk[j, k] = expectation_val
+    return matrix_jk
+
+
+def GST(
+    gate_set=Union[tuple, set, list],
+    nshots=int(1e4),
+    noise_model=None,
+    include_empty=False,
+    invert_register=None,
+    Pauli_Liouville=False,
+    backend=None,
+):
+    """This is a wrapper function that runs gate set tomography for a list of gates. One can choose to output the gate set tomography
+    for each gate in the Pauli-Liouville representation or not.
+
+
+    Args:
+        gate_set (tuple, set, list): A list containing :class:`qibo.gates.abstract.Gate`.
+        nshots (int, optional): Number of shots used in Gate Set Tomography.
+        noise_model (:class:`qibo.noise.NoiseModel`, optional): Noise model applied to simulate noisy computation.
+        include_empty (bool, optional): If ``False``, only perform gate set tomography for the list of gates in ``gate_set``.
+            If ``True``, perform gate set tomography for the list of gates in ``gate_set`` and also empty circuits.
+        invert_register (bool, optional): If ``True``, one needs to specify which qubit(s) to reset.
+        Pauli_Liouville (bool, optional): If ``True``, returns gate set tomography of the gates in the Pauli-Liouville representation.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): Calculation engine.
+    Returns:
+        list(ndarray): List of matrices of the gate(s) in gate set tomography.
+    """
+    matrices = []
+    empty_matrices = []
+    for nqubits in range(1, 3):
+        empty_matrix = _gate_set_tomography(
+            nqubits=nqubits,
+            gate=None,
+            nshots=nshots,
+            noise_model=noise_model,
+            invert_register=invert_register,
+            backend=backend,
+        )
+        empty_matrices.append(empty_matrix)
+        if include_empty:
+            matrices.append(empty_matrix)
```

Comment:
```suggestion
    if include_empty or Pauli_Liouville:
        for nqubits in range(1, 3):
            empty_matrix = _gate_set_tomography(
                nqubits=nqubits,
                gate=None,
                nshots=nshots,
                noise_model=noise_model,
                invert_register=invert_register,
                backend=backend,
            )
            empty_matrices.append(empty_matrix)
```
I would run this only if necessary, namely only if the user asks for the empty matrices or the pauli liouville representation.